pinery-lims/src/main/java/ca/on/oicr/pinery/lims/LimsAttribute.java

```diff
@@ -77,21 +78,21 @@ public String asString(boolean useEnumToString) {
     private static void validate() {
         Set<String> ins = new HashSet<>();
         Set<String> outs = new HashSet<>();
-        for (SampleAttribute sa : SampleAttribute.values()) {
-            if (Sets.intersection(ins, sa.in).isEmpty()) {
-                ins.addAll(sa.in);
+        for (LimsAttribute la : LimsAttribute.values()) {
+            if (Sets.intersection(ins, la.in).isEmpty()) {
+                ins.addAll(la.in);
             } else {
-                throw new RuntimeException("Duplicate: [" + sa.in.toString() + "]");
+                throw new RuntimeException("Duplicate: [" + la.in.toString() + "]");
             }
-            if (!ins.contains(sa.out)) {
-                ins.add(sa.out);
+            if (!ins.contains(la.out)) {
+                ins.add(la.out);
             } else {
-                throw new RuntimeException("Duplicate: [" + sa.out.toString() + "]");
+                throw new RuntimeException("Duplicate: [" + la.out.toString() + "]");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "out" is already a string, there's no need to call "toString()" on it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1858)